const foo: <A extends number>(x: A) => Promise<A, "RangeError">;
